docs/release.md

```diff
@@ -0,0 +1,68 @@
+# OSA Release process
+
+OSA Release process consist of multiple dimensions:
+* Plugin code release
+* Config release
+* VM image release
+
+Update might be one or any combination of all dimensions.
+Plugin code release is a new branch and considered a major release. Any VM image or config change is a minor release.
+
+# Versioning
+
+## Major releases
+
+Major release creation flow:
+1. Create a branch with a plugin code you wish to release.
+2. Commit to update `PluginConfig` referencing release image versions.
+3. Git tag major release version (`v3.0`)
+4. Build release images (sync, metricsbridge, etc)
+5. Make sure VM image is released and is available to be used by plugin code.
+
+## Minor releases
+
+If there is a need to update already released code - minor release is done.
+This might happen if we make a mistake, want to update configuration, or not code related fix.
+
+Minor release creation flow:
+1. Add commit to release branch with required changes.
+2. Git tag minor version (`v3.1`)
+3. Build release images (sync, metricsbridge, etc)
+
+Branching model:
+```
+-------master--*----------*-----------------------
+                \          \
+                 \-release-v3------T(v3.0)----
+                             \-release-v4----T(v4.0)-----T(v4.1)-
```

Comment:
this doesn't make a lot of sense to me, maybe it's just the graphic.
imho major releases should be made from master not the previous release.

```
-------master--*----------*-----------------------
               \          \
                \           \-release-v4----T(v4.0)-----T(v4.1)-
                 \-release-v3------T(v3.0)----

```

Reply:
yes, its graphic,  it is as you pained it :) will use your ASCII :)